ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/InsightsQueryTest.kt

```diff
@@ -0,0 +1,234 @@
+                }
+
+                step("Verify new query settings have persisted") {
+                    openInsightsQueryDialogFromResults()
+                    find<JTextFieldFixture>(byXpath("//div[@class='JFormattedTextField' and @visible_text='35']"))
+                    assertThat(find<ComboBoxFixture>(byXpath("//div[@class='ComboBox']")).selectedText()).isEqualTo("Hours")
+                }
+            }
+        }
+    }
+
+    @AfterAll
+    fun cleanup() {
+        try {
+            val client = CloudWatchLogsClient.create()
+            client.verifyDeletedLogGroup(logGroupName)
+        } catch (e: Exception) {
+            log.error("Unable to remove log groups", e)
+        }
+    }
+
+    private fun RemoteRobot.openInsightsQueryDialogFromExplorer(groupName: String) = step("Open insights query dialog") {
+        awsExplorer {
+            openExplorerActionMenu(cloudWatchExplorerLabel, groupName)
+        }
+        find<ComponentFixture>(byXpath("//div[@accessiblename='Open Query Editor']")).click()
+    }
+
+    private fun ContainerFixture.openInsightsQueryDialogFromResults() = step("Open query editor") {
+        findAndClick("//div[@text='Open Query Editor']")
+    }
+
+    private fun CloudWatchLogsClient.verifyDeletedLogGroup(logGroup: String) {
+        try {
+            deleteLogGroup {
+                it.logGroupName(logGroupName)
+            }
+        } catch (e: ResourceNotFoundException) {
+            log.info("Log group $logGroup did not exist")
+            return
+        } catch (e: Exception) {
+            log.error("DeleteLogGroup returned an error, cannot attempt deletion again", e)
+            return
+        }
+        log.info("Log group deleted: $logGroup")
+    }
+
+    private suspend fun CloudWatchLogsClient.waitForResults(logGroup: String) {
+        withTimeout(Duration.ofMinutes(5).toMillis()) {
+            while (true) {
+                val stopTime = Instant.now()
+                val startTime = stopTime.minus(Duration.ofHours(1))
+                val queryId = startQuery {
+                    it.logGroupName(logGroup)
+                    it.startTime(startTime.toEpochMilli())
+                    it.endTime(stopTime.toEpochMilli())
+                    it.queryString(DEFAULT_INSIGHTS_QUERY_STRING)
+                }.queryId()
+                log.info("started query with id: $queryId")
+
+                while (true) {
+                    val result = getQueryResults {
+                        it.queryId(queryId)
+                    }
+
+                    if (result.results().size >= 2) {
+                        log.info("query returned enough results, continuing")
+                        return@withTimeout
+                    } else if (result.status() in setOf(QueryStatus.FAILED, QueryStatus.CANCELLED)) {
+                        throw IllegalStateException("Reached terminal condition while waiting for log propagation")
```

Comment:
we should log the result and reason here, otherwise we won't know what it returned